scipy/signal/_spline_filters.py

```diff
@@ -644,3 +647,161 @@ def qspline1d_eval(cj, newx, dx=1.0, x0=0):
         result += cj[indj] * _quadratic(newx - thisj)
     res[cond3] = result
     return res
+
+
+def symiirorder1(signal, c0, z1, precision=-1.0):
+    """
+    Implement a smoothing IIR filter with mirror-symmetric boundary conditions
+    using a cascade of first-order sections.
+
+    The second section uses a reversed sequence.  This implements a system with
+    the following transfer function and mirror-symmetric boundary conditions::
+
+                           c0
+           H(z) = ---------------------
+                   (1-z1/z) (1 - z1 z)
+
+    The resulting signal will have mirror symmetric boundary conditions
+    as well.
+
+    Parameters
+    ----------
+    input : ndarray
+        The input signal. If 2D, then the filter will be applied in a batched
+        fashion across the last axis.
+    c0, z1 : scalar
+        Parameters in the transfer function.
+    precision :
+        Specifies the precision for calculating initial conditions
+        of the recursive filter based on mirror-symmetric input.
+
+    Returns
+    -------
+    output : ndarray
+        The filtered signal.
+    """
+    if np.abs(z1) >= 1:
+        raise ValueError('|z1| must be less than 1.0')
+
+    if signal.ndim > 2:
+        raise ValueError('Input must be 1D or 2D')
+
+    squeeze_dim = False
+    if signal.ndim == 1:
+        signal = signal[None, :]
+        squeeze_dim = True
+
+    if np.issubdtype(signal.dtype, np.integer):
+        signal = signal.astype(np.promote_types(signal.dtype, np.float32))
+
+    y0 = symiirorder1_ic(signal, z1, precision)
+
+    # Apply first the system 1 / (1 - z1 * z^-1)
+    b = np.ones(1, dtype=signal.dtype)
+    a = np.r_[1, -z1]
+    a = a.astype(signal.dtype)
+
+    # Compute the initial state for lfilter.
+    zii = y0 * z1
+
+    y1, _ = lfilter(b, a, axis_slice(signal, 1), zi=zii)
+    y1 = np.c_[y0, y1]
+
+    # Compute backward symmetric condition and apply the system
+    # c0 / (1 - z1 * z)
+    b = np.asarray([c0], dtype=signal.dtype)
+    out_last = -c0 / (z1 - 1.0) * axis_slice(y1, -1)
```

Comment:
I cannot see how this computation lines up with the comment. However `y1` is calculated, it's really not obvious to me how multiplying into a subtraction while keeping the `1 -` constant should work.

Reply:
This is not me, this PR just moved this line.

Reply:
The `-c0 / (z1 - 1)` corresponds to the initial state when the filter is run in backwards, this sample will replace the last value of the output. From my understanding, this initial value is the steady state for the filter when ran in backwards. 

This implementation detail was present in the original C template file.

Reply:
@andfoy would be able to come up with a good wording?

Reply:
```suggestion
    # The initial symmetric condition of the system c0 / (1 - z1 * z)
    # corresponds to c0 * y[N - 1] * \sum_{n = 0}^{\infty}  z1^{n},
    # since |z1| < 1.0 , it converges to -c0 / (z1 - 1) * y[N - 1]
    b = np.asarray([c0], dtype=signal.dtype)
    out_last = -c0 / (z1 - 1.0) * axis_slice(y1, -1)
```